tests/test_revisions.py

```diff
@@ -0,0 +1,124 @@
+        "https://mozphab.dev.mozaws.net/p/imadueme_admin/",
+        "username":
+        "imadueme_admin"
+    },
+    "bug_id":
+    None,
+    "date_created":
+    1495217456,
+    "date_modified":
+    1498565006,
+    "id":
+    1,
+    "parent_revisions": [
+        {
+            "author": {
+                "image_url":
+                "https://d2kb8dptaglwte.cloudfront.net/file/data/ifecwljgvarrzk2z3psj/PHID-FILE-2pj3vvuhi6gof6susyqu/alphanumeric_lato-dark_M.png-_bdb0f0-255%2C255%2C255%2C0.4.png",
+                "phid":
+                "PHID-USER-uhfcvek26nuzv2wdf7yp",
+                "real_name":
+                "Mark Cote",
+                "url":
+                "https://mozphab.dev.mozaws.net/p/mcote_admin/",
+                "username":
+                "mcote_admin"
+            },
+            "bug_id": None,
+            "date_created": 1496179554,
+            "date_modified": 1496697530,
+            "id": 5,
+            "parent_revisions": [],
+            "phid": "PHID-DREV-ixfdpmt54f5tz2gf6i7m",
+            "repo": {
+                "full_name": "rMOZILLACENTRAL mozilla-central",
+                "phid": "PHID-REPO-khtgyhvsqje2yylf2hzd",
+                "short_name": "rMOZILLACENTRAL",
+                "url": "https://mozphab.dev.mozaws.net/source/mozilla-central/"
+            },
+            "status": 0,
+            "status_name": "Needs Review",
+            "summary": "",
+            "test_plan": "",
+            "title": "Test manual rename.",
+            "url": "https://mozphab.dev.mozaws.net/D5"
+        }
+    ],
+    "phid":
+    "PHID-DREV-kcobr5ckqlshytrnxnos",
+    "repo": {
+        "full_name": "rMOZILLACENTRAL mozilla-central",
+        "phid": "PHID-REPO-khtgyhvsqje2yylf2hzd",
+        "short_name": "rMOZILLACENTRAL",
+        "url": "https://mozphab.dev.mozaws.net/source/mozilla-central/"
+    },
+    "status":
+    0,
+    "status_name":
+    "Needs Review",
+    "summary":
+    "This is a diff",
+    "test_plan":
+    "Look at it",
+    "title":
+    "Bug 564987 - buildbot should cope when firefox/plugins doesn't exist when the config option is just right to trigger a super long revision name that can be used to test the ui",
+    "url":
+    "https://mozphab.dev.mozaws.net/D1"
+}
+
+
+def test_render_valid_revision(client, monkeypatch):
+    monkeypatch.setenv('LANDO_API_URL', 'http://landoapi.test')
+    with requests_mock.mock() as m:
+        m.get('http://landoapi.test/revisions/D1', json=LANDO_API_RESPONSE)
+        response = client.get('/revisions/D1')
+    assert response.status_code == 200
+
+
+def test_missing_revision_returns_404(client, monkeypatch):
+    monkeypatch.setenv('LANDO_API_URL', 'http://landoapi.test')
+    with requests_mock.mock() as m:
+        m.get('http://landoapi.test/revisions/D1057', status_code=404)
+        response = client.get('/revisions/D1057')
+    assert response.status_code == 404
+
+
+def test_lando_api_timeout_returns_504(client, monkeypatch):
```

Comment:
This, the 503, and the 502 test case aren't needed. A generic 500 test is needed.